bokeh/models/plots.py

```diff
@@ -344,11 +343,15 @@ def add_glyph(self, source_or_glyph: Glyph | ColumnarDataSource, glyph: Glyph |
         self.renderers.append(g)
         return g
 
-    def add_tile(self, tile_source: TileSource, **kwargs: Any) -> TileRenderer:
+    def add_tile(self, tile_source: Union[TileSource, xyzservices.TileProvider, str], retina=False, **kwargs: Any) -> TileRenderer:
```

Comment:
We prefer the new 3.10 syntax `TileSource | xyzservices.TileProvider | str` (where ever lazy annotations are permitted). Also `retina` needs an explicit type.

```suggestion
    def add_tile(self, tile_source: TileSource | xyzservices.TileProvider | str, retina: bool = False, **kwargs: Any) -> TileRenderer:
```